app/models/ability.rb

```diff
@@ -0,0 +1,10 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :manage, :all if user.role == 'admin'
+
+    can :manage, Post, author_id: user.id
+    can :manage, Comment, author_id: user.id
+  end
```

Comment:
- [ ] **[OPTIONAL]** While you have added the ability to manage resources only by the author and admin, I don't think it is a good idea to restrict a user from visiting the show page.

  You can add the following spinnet to allow users view page:
  ```
  can :read, :all
  ```

  ![image](https://github.com/Fahim-Khan-P/blog/assets/36542986/d6e802a5-9141-4a4b-a706-723b8323d01d)